systemtest/src/test/java/io/strimzi/systemtest/kafka/TieredStorageST.java

```diff
@@ -0,0 +1,159 @@
+import io.strimzi.systemtest.resources.NamespaceManager;
+import io.strimzi.systemtest.resources.NodePoolsConverter;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.imageBuild.ImageBuild;
+import io.strimzi.systemtest.resources.minio.SetupMinio;
+import io.strimzi.systemtest.storage.TestStorage;
+import io.strimzi.systemtest.templates.crd.KafkaNodePoolTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaTopicTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.test.TestUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static io.strimzi.systemtest.TestConstants.REGRESSION;
+import static io.strimzi.systemtest.TestConstants.TIERED_STORAGE;
+
+/**
+ * @description This test suite covers scenarios for Tiered Storage integration implemented within Strimzi.
+ *
+ * @steps
+ *  1. - Create test namespace
+ *  2. - Build Kafka image based on passed parameters like image full name, base image, Dockerfile path (via Kaniko or OpenShift build)
+ *  3. - Deploy Minio in test namespace and init the client inside the Minio pod
+ *  4. - Init bucket in Minio for purposes of these tests
+ *  5. - Deploy Strimzi Cluster Operator
+ *
+ * @usecase
+ *  - tiered-storage-integration
+ */
+@Tag(REGRESSION)
+@Tag(TIERED_STORAGE)
+public class TieredStorageST extends AbstractST {
+    private static final Logger LOGGER = LogManager.getLogger(TieredStorageST.class);
+
+    private static final String IMAGE_NAME = "kafka-tiered-storage";
+    private static final String TIERED_STORAGE_DOCKERFILE = TestUtils.USER_PATH + "/../systemtest/src/test/resources/tiered-storage/Dockerfile";
+    private static final String BUCKET_NAME = "test-bucket";
+    private static final String BUILT_IMAGE_TAG = "latest";
+
+    /**
+     * @description This testcase is focused on testing of Tiered Storage integration implemented within Strimzi.
+     * The tests use Aiven Tiered Storage plugin - <a href="https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/tree/main">...</a>
+     *
+     * @steps
+     *  1. - Deploys KafkaNodePool resource with Broker NodePool with PV of size 10Gi
+     *  2. - Deploys Kafka resource with configuration of Tiered Storage for Aiven plugin, pointing to Minio S3, and with image built in beforeAll
+     *  3. - Creates topic with enabled Tiered Storage sync with size of segments set to 10mb (this is needed for speedup the sync)
+     *  4. - Starts continuous producer to send data to Kafka
+     *  5. - Wait until Minio size is not empty (contains data from Kafka)
+     *
+     * @usecase
+     *  - tiered-storage-integration
+     */
+    @ParallelTest
+    void testTieredStorageWithAivenPlugin() {
```

Comment:
So I see the purpose of this test case (i.e., checking if aiven plugin works with our deployment) but should we also test the Tiered functionality or we simply let it to Kafka? For instance that we would have two layers of storage (hot and cold and we would somehow configure it and checks if it works as we expect? Or maybe this could be implemented in our long-running env right?

Reply:
Well the test case does exactly this. There is persistent volume for Kafka configured by Strimzi and also tiered storage plugin config done by Strimzi for Aiven plugin. The test case setup everything, send messages to Kafka and wait until the messages will be sent to tiered storage (minio in that case). Then all messages are consumed by a consumer - some messages are on PV because are not synced yet and some are only in tiered storage because they were already synced. Do you have in mind any different strategy than this one? Or is my understating of the functionality wrong?

Reply:
Oh, I see, maybe it would be good to better document it or maybe I am dumb 🤷 :D . In the verify phase you simply check both sides (i.e., hot (PV) and cold storage (Minio)) for me it was quite hidden, but after your explanation it makes sense. 

One question...Would it be good to check specific data written in cold and hot storage to prove integrity? For instance, if the producer sends ("hello") we should check that both PV (hot) and Minio (cold) have such data. WDYT?

Reply:
Isn't it something that should be tested on plugin/Kafka side? We basically care about Strimzi provided configuration. I guess I can add it if there is a mechanism in STs for that kind of check, but I don't see much value from Strimzi pov. But it is just my opinion.

Reply:
> Isn't it something that should be tested on plugin/Kafka side? We basically care about Strimzi provided configuration. I guess I can add it if there is a mechanism in STs for that kind of check, but I don't see much value from Strimzi pov. But it is just my opinion.

@scholzj wdyt?

Reply:
Discussed offline and agreed we will keep the test as it is.